### DISCUSSION TITLE
蓝盾自定义插件在构建机上的安装问题

### DISCUSSION THREAD

**Discussion by isseptember** [Tencent/bk-ci#6836] [Q&A]
蓝盾社区版 v1.2.11

python开发蓝盾插件，task.json里demands如下。麻烦帮忙解答以下问题，谢谢~
1、在流水线里执行时，除去第一次，都会先卸载再安装吗？
2、插件安装在构建机上是全局的吗？
3、构建机设置最大构建并发数为30，同一流水线隔5s构建一次，一共10次，个别会出现导包失败的问题，是否是多个任务同时在卸载又安装导致的呢？应该怎么避免这种情况呢？
4、demands的值为非必填，但是不填，执行插件时就不会主动安装了？是否可以通过此处设置插件不卸载，只升级呢？

![image](https://user-images.githubusercontent.com/54738906/170430350-b825567f-5017-492a-82a6-81c3a98d342e.png)
![image](https://user-images.githubusercontent.com/54738906/170430382-66b2370e-15c3-41a6-86e2-ef68607cc685.png)

**Comment by irwinsun** [comment]
哥们， python是有这个并发问题，因为没有使用pyenv这类虚拟环境， 所以是全局的。
